tests/foreman/installer/test_infoblox.py

```diff
@@ -15,183 +17,162 @@
+        ).status
+        == 0
+    )
+    assert (
+        module_target_sat.execute(
+            'satellite-installer --help | grep foreman-proxy-dns-provider | grep infoblox'
+        ).status
+        == 0
+    )
+    assert (
+        module_target_sat.execute(
+            f'satellite-installer --help | grep foreman-proxy-dhcp-server | '
+            f'grep {settings.infoblox.hostname}'
+        ).status
+        == 0
+    )
+    assert (
+        module_target_sat.execute(
+            f'satellite-installer --help | grep foreman-proxy-plugin-dns-infoblox-dns-server | '
+            f'grep {settings.infoblox.hostname}'
+        ).status
+        == 0
+    )
+
+    macaddress = gen_mac(multicast=False)
+    # using the domain name as defined in Infoblox DNS
+    domain = module_target_sat.api.Domain(
+        name='vms.sat.rdu2.redhat.com',
+        location=[module_location],
+        dns=module_provisioning_capsule.id,
+        organization=[module_sca_manifest_org],
+    ).create()
+    subnet = module_target_sat.api.Subnet(
+        location=[module_location],
+        organization=[module_sca_manifest_org],
+        network=settings.infoblox.network,
+        cidr=settings.infoblox.network_prefix,
+        mask=settings.infoblox.netmask,
+        from_=settings.infoblox.start_range,
+        to=settings.infoblox.end_range,
+        boot_mode='DHCP',
+        ipam='DHCP',
+        dhcp=module_provisioning_capsule.id,
+        tftp=module_provisioning_capsule.id,
+        dns=module_provisioning_capsule.id,
+        discovery=module_provisioning_capsule.id,
+        remote_execution_proxy=[module_provisioning_capsule.id],
+        domain=[domain.id],
+    ).create()
+    host = module_target_sat.api.Host(
+        organization=module_sca_manifest_org,
+        location=module_location,
+        name=gen_string('alpha').lower(),
+        mac=macaddress,
+        operatingsystem=module_sync_kickstart_content.os,
+        architecture=default_architecture,
+        domain=domain,
+        subnet=subnet,
+        root_pass=settings.provisioning.host_root_password,
+        ptable=default_partitiontable,
+        content_facet_attributes={
+            'content_source_id': module_provisioning_capsule.id,
+            'content_view_id': module_default_org_view.id,
+            'lifecycle_environment_id': module_lce_library.id,
+        },
+    ).create()
+    # check if A Record is created for the host IP on Infoblox
+    url = f"https://{settings.infoblox.hostname}/wapi/v2.0/ipv4address?ip_address={host.ip}"
+    auth = (settings.infoblox.username, settings.infoblox.password)
+    result = requests.get(url, auth=auth, verify=False)
+    assert result.status_code == 200
+    # check hostname and ip is present in A record
+    assert host.name in result.text
+    assert host.ip in result.text
+
+    @request.addfinalizer
+    def _finalize():
+        module_target_sat.api.Subnet(id=subnet.id, domain=[]).update()
```

Comment:
is subnet update required in teardown?

Reply:
Yes, the host cannot be deleted directly as the domain is attached to it .